### PR TITLE
Adds PulpPluginAppConfig.version warnings

### PR DESCRIPTION
**Pull request by bmbouter**
Starting with 3.10 plugins will be required to declare their version as
the `version` attribute on the PulpPluginAppConfig subclass. This adds
warnings starting in 3.9 encouraging plugin writers to do so.

Additionally, it sets the `version` on PulpAppConfig so that it doesn't
complain about itself. The bumpversion spec also will bump this string
at release time.

closes #6671

### CODE REVIEW

File: docs/plugins/reference/how-plugins-work.rst [hunk 1]
```diff
@@ -8,19 +8,22 @@ Plugin Django Application
 
 Like the Pulp Core itself, all Pulp Plugins begin as Django Applications, started like any other
 with `pulpcore-manager startapp <your_plugin>`. However, instead of subclassing Django's
-`django.apps.AppConfig` as seen `in the Django documentation
-<https://docs.djangoproject.com/en/1.8/ref/applications/#for-application-authors>`_, Pulp Plugins
-identify themselves as plugins to the Pulp Core by subclassing
-:class:`pulpcore.plugin.PulpPluginAppConfig`
+`django.apps.AppConfig` as seen `in the Django documentation <https://docs.djangoproject.com/en/2.2
+/ref/applications/#for-application-authors>`_, Pulp Plugins identify themselves as plugins to
+pulpcore by subclassing :class:`pulpcore.plugin.PulpPluginAppConfig`
```
Comment (daviddavis) [inline docs/plugins/reference/how-plugins-work.rst:13]:
```suggestion
pulpcore by subclassing :class:`pulpcore.plugin.PulpPluginAppConfig`.
```

File: .bumpversion.cfg [hunk 2]
```diff
@@ -14,6 +14,6 @@ values =
 	dev
 	prod
 
-[bumpversion:file:./pulpcore/__init__.py]
+[bumpversion:file:./pulpcore/app/apps.py]
```
Comment (daviddavis) [inline .bumpversion.cfg:17]: Don't forget to update https://github.com/pulp/plugin_template/blob/aa179c72a3158c70a1b93077260da4a2a7a5eab2/templates/bootstrap/.bumpversion.cfg.j2#L17

**Review by daviddavis — APPROVED** [review]
LGTM.

File: .bumpversion.cfg (hunk 2 above)
Comment (bmbouter) [reply .bumpversion.cfg:17]: https://github.com/pulp/plugin_template/pull/310

**Review by dralley — APPROVED** [review]

File: .ci/scripts/release.py [hunk 3]
```diff
@@ -177,14 +181,17 @@ def validate_redmine_data(redmine_query_url, redmine_issues):
...
-version = {}
-with open(f"{plugin_path}/{plugin_name}/__init__.py") as fp:
-    version_line = [line for line in fp.readlines() if "__version__" in line][0]
-    exec(version_line, version)
-new_dev_version = version["__version__"]
+new_dev_version = None
+with open(f"{plugin_path}/setup.py") as fp:
+    for line in fp.readlines():
+        if "version=" in line:
+            version = line.split('"')[1]
```
Comment (bmbouter) [inline .ci/scripts/release.py:188]: This should be `new_dev_version` I don't understand why it didn't raise?

**Pull request merged**